skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/BaseSkaffoldSettingsEditor.kt

```diff
@@ -42,17 +44,23 @@ open class BaseSkaffoldSettingsEditor : SettingsEditor<RunConfiguration>() {
         return basePanel
     }
 
-    override fun applyEditorTo(runConfig: RunConfiguration) {
+    override fun applyEditorTo(runConfig: BaseSkaffoldRunConfiguration) {
         val selectedSkaffoldFile: VirtualFile =
             skaffoldFilesComboBox.getItemAt(skaffoldFilesComboBox.selectedIndex)
                 ?: throw ConfigurationException(message("skaffold.no.file.selected.error"))
 
         if (!SkaffoldFileService.instance.isSkaffoldFile(selectedSkaffoldFile)) {
             throw ConfigurationException(message("skaffold.invalid.file.error"))
         }
+
+        // save properties
+        runConfig.skaffoldConfigurationFilePath = selectedSkaffoldFile.path
     }
 
-    override fun resetEditorFrom(runConfig: RunConfiguration) {
+    override fun resetEditorFrom(runConfig: BaseSkaffoldRunConfiguration) {
         skaffoldFilesComboBox.setProject(runConfig.project)
+        runConfig.skaffoldConfigurationFilePath?.let {
```

Comment:
just to make sure I understand this snippet:

if there is a skaffold path set, you locate the corresponding `File` on the filesystem, then set the file in the combobox model?

Reply:
Correct, regardless of whether this file exists or not in the filesystem anymore, its path is taken, converted into VirtualFile and shown in the combo box.